src/main.rs

```diff
@@ -68,96 +79,275 @@ async fn main() {
+                            Some(path.to_owned().into_path_buf()),
+                            Some(resource.id),
+                            None,
+                        ),
+                        EntryOutput::Path => {
+                            (Some(path.to_owned().into_path_buf()), None, None)
+                        }
+                        EntryOutput::Id => (None, Some(resource.id), None),
+                        EntryOutput::Link => {
+                            let mut file = File::open(path).unwrap();
+                            let mut contents = String::new();
+                            file.read_to_string(&mut contents).unwrap();
+                            (None, None, Some(contents))
+                        }
+                    };
+
+                    StorageEntry {
+                        path,
+                        resource,
+                        content,
+                        tags,
+                        scores,
+                        datetime,
+                    }
                 })
                 .collect::<Vec<_>>();
 
             match sort {
-                Some(Sort::Asc) => resources
-                    .sort_by(|(_, _, _, _, a), (_, _, _, _, b)| a.cmp(b)),
+                Some(Sort::Asc) => {
+                    storage_entries.sort_by(|a, b| a.datetime.cmp(&b.datetime))
+                }
 
-                Some(Sort::Desc) => resources
-                    .sort_by(|(_, _, _, _, a), (_, _, _, _, b)| b.cmp(a)),
+                Some(Sort::Desc) => {
+                    storage_entries.sort_by(|a, b| b.datetime.cmp(&a.datetime))
+                }
                 None => (),
             };
 
             if let Some(filter) = filter {
-                resources = resources
-                    .into_iter()
-                    .filter(|(_, _, tags_list, _, _)| {
-                        tags_list
-                            .split(',')
-                            .any(|tag| tag.trim() == filter)
-                    })
-                    .collect();
+                storage_entries.retain(|entry| {
+                    entry
+                        .tags
+                        .as_ref()
+                        .map(|tags| tags.contains(filter))
+                        .unwrap_or(false)
+                });
             }
 
-            for (path, resource, tags_list, scores_list, datetime) in resources
-            {
-                let mut output = String::new();
+            let no_tags = "NO_TAGS";
+            let no_scores = "NO_SCORE";
+
+            let longest_path = storage_entries
+                .iter()
+                .map(|entry| {
+                    if let Some(path) = entry.path.as_ref() {
+                        path.display().to_string().len()
+                    } else {
+                        0
+                    }
+                })
+                .max_by(|a, b| a.cmp(b))
+                .unwrap_or(0);
```

Comment:
I understand that the current approach attempts to manually format the output as a table.
However, I think the code for `Command::List` is now so long (also considering this is part of `main.rs`

On top of my head, I can think of two alternative approaches, but I'm open to other suggestions.
1. Dedicated struct:
    - Define a new struct encapsulating the list data and implement the `Display` trait for formatted output
    - In `main.rs`, create an instance of this struct and simply print it. For the current setup, this could be as easy as defining a type `Vec<StorageEntry>` and implementing `Display` for it
2. Use an existing crate to handle table formatting effectively. I can recommend [`radicle-term`](https://crates.io/crates/radicle-term).